Meadowlark-js/backends/meadowlark-postgres-backend/src/repository/Db.ts

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import { Pool, Client } from 'pg';
+import { Logger } from '@edfi//meadowlark-core';
+import format from 'pg-format';
+
+let dbPool: Pool | null = null;
+
+const dbConfiguration = {
+  host: process.env.POSTGRES_HOST ?? 'localhost',
+  port: Number(process.env.POSTGRES_PORT ?? 5555),
+  user: process.env.POSTGRES_USER ?? 'postgres',
+  password: process.env.POSTGRES_PASSWORD ?? 'abcdefgh1!',
+  database: 'meadowlark',
+};
+
+export async function checkExistsAndCreateTables(client: Client) {
+  try {
+    await client.query(`CREATE SCHEMA IF NOT EXISTS meadowlark`);
+
+    const documentsTableSql =
+      'CREATE TABLE IF NOT EXISTS meadowlark.documents(' +
+      '_pk bigserial PRIMARY KEY,' +
```

Comment:
Do we need this? Our document id is unique and identifying, so could it be the pk?

Reply:
Otherwise, we'd also need an index on id

Reply:
I haven't yet looked to see if there is a definitive answer, and I think it verges on "religious war." Personally I prefer to have a synthetic numeric primary key in a relational database because of advantages with storage size and sorting behavior. Yes, will therefore need an index on our natural key id in that case.

Reply:
In a typical relational model I prefer surrogate keys. The complexity of the ODS/API is a good example of why.  I'll bet you could greatly simplify the ODS/API code if the DB used surrogate keys instead of cascading composite key columns on to sub-tables. I know MetaEd generation would be significantly simpler. We certainly wouldn't have needed the "merge" keyword, with all its issues and complexities.

The reason I'm thinking we don't need surrogate keys is that this isn't a relational model - at least not yet. We don't need to join the references table back to the documents table. That would over complicate things and be needlessly slower. All we need the reference table for is delete reference checking. That's just a simple query of the table for any row with a matching references_to entry. Any results means the delete should be failed.

It's entirely possible that we'll want a relational model between the documents table and some future security tables, but that's something we can figure out later. Until then, it would be nice to keep things simple.